packages/perspective-viewer-d3fc/src/js/plugin/template.js

```diff
@@ -27,7 +28,7 @@ class D3FCChartElement extends HTMLElement {
         this.remove();

```

Comment:
I think at this point, we should compare `crossValues`, `mainValues` and `splitValues`, and if there are no differences, don't call `this.remove()`, and instead do:

````javascript
this._settings = {...this._settings, data: settings.data}
````

That should preserve `all` proprties of the settings (including the hidden values), while just passing in the updated data, and also avoids removing and recreating the DOM.

Reply:
Would we still be calling `this.draw()`?

Reply:
Yep.
D3 should update the existing DOM elements rather than have to recreate from scratch.